### ISSUE TITLE
GET /api/trpc/authCallback?batch=1&input=%7B%7D 500 in 358ms

### ISSUE THREAD

**Issue by JatinSri1909** [joschan21/quill#111]
After entering emailid and otp, the loader with the text 'setting up your account' continuously shown in the page. In the console , this message is shown: GET /api/trpc/authCallback?batch=1&input=%7B%7D 500 in 358ms

**Comment by JatinSri1909** [comment]
> After entering emailid and otp, the loader with the text 'setting up your account' continuously shown in the page. In the console , this message is shown: GET /api/trpc/authCallback?batch=1&input=%7B%7D 500 in 358ms

![Screenshot 2024-10-18 223023](https://github.com/user-attachments/assets/048b243a-9a60-4842-8ce1-9e3df6d9547c)
![Uploading Screenshot 2024-10-18 223054.png…]()

**Comment by coolchigi** [comment]
I managed to fix the error in mine! 

Couple of places you could check:

`app/dashboard/page.tsx` -> ensure you are returning the right component. 

For me, I was returning an undefined dashboard component. I changed it to `return <div>{user.email}</div>` & that seemed to fix the error

**Comment by JatinSri1909** [comment]
Nope, It didn't work. I get this error : " GET /api/trpc/authCallback?batch=1&input=%7B%7D 500 in 512ms"

**Comment by coolchigi** [comment]
Okay, here's another thing to try:

Await the user. I remember seeing an error in my console about `getUser` returning a Promise.  
```javascript
const user = await getUser();

if (!user || !user.id)
```

**Comment by JatinSri1909** [comment]
Done. Thanks for the help

**Comment by coolchigi** [comment]
Did that fix the error?

**Comment by JatinSri1909** [comment]
Yes
